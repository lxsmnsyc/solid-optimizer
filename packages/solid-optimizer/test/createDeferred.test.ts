import { describe, expect, it } from 'vitest';
import compile from './compile';

describe('createDeferred', () => {
  describe('Expression arguments', () => {
    it('should work for ImportSpecifier', async () => {
      expect(await compile(`
        import { createDeferred } from 'solid-js';
        
        createDeferred(() => update());
      `)).toMatchSnapshot();
    });
    it('should work for aliased ImportSpecifier', async () => {
      expect(await compile(`
        import { createDeferred as x } from 'solid-js';
        
        x(() => update());
      `)).toMatchSnapshot();
    });
    it('should work for aliased string ImportSpecifier', async () => {
      expect(await compile(`
        import { 'createDeferred' as x } from 'solid-js';
        
        x(() => update());
      `)).toMatchSnapshot();
    });
    it('should work for namespace', async () => {
      expect(await compile(`
        import * as solid from 'solid-js';
        
        solid.createDeferred(() => update());
      `)).toMatchSnapshot();
    });
  });
  describe('spreads', () => {
    it('should work for ImportSpecifier', async () => {
      expect(await compile(`
        import { createDeferred } from 'solid-js';
        
        createDeferred(...example);
      `)).toMatchSnapshot();
    });
    it('should work for aliased ImportSpecifier', async () => {
      expect(await compile(`
        import { createDeferred as x } from 'solid-js';
        
        x(...example);
      `)).toMatchSnapshot();
    });
    it('should work for aliased string ImportSpecifier', async () => {
      expect(await compile(`
        import { 'createDeferred' as x } from 'solid-js';
        
        x(...example);
      `)).toMatchSnapshot();
    });
    it('should work for namespace', async () => {
      expect(await compile(`
        import * as solid from 'solid-js';
        
        solid.createDeferred(...example);
      `)).toMatchSnapshot();
    });
  });
});
