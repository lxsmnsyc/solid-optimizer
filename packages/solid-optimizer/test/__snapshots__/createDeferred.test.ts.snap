// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createDeferred > Expression arguments > should work for ImportSpecifier 1`] = `
"import { createDeferred } from 'solid-js';
() => update();"
`;

exports[`createDeferred > Expression arguments > should work for aliased ImportSpecifier 1`] = `
"import { createDeferred as x } from 'solid-js';
() => update();"
`;

exports[`createDeferred > Expression arguments > should work for aliased string ImportSpecifier 1`] = `
"import { 'createDeferred' as x } from 'solid-js';
() => update();"
`;

exports[`createDeferred > Expression arguments > should work for namespace 1`] = `
"import * as solid from 'solid-js';
() => update();"
`;

exports[`createDeferred > spreads > should work for ImportSpecifier 1`] = `
"import { createDeferred } from 'solid-js';
example[0];"
`;

exports[`createDeferred > spreads > should work for aliased ImportSpecifier 1`] = `
"import { createDeferred as x } from 'solid-js';
example[0];"
`;

exports[`createDeferred > spreads > should work for aliased string ImportSpecifier 1`] = `
"import { 'createDeferred' as x } from 'solid-js';
example[0];"
`;

exports[`createDeferred > spreads > should work for namespace 1`] = `
"import * as solid from 'solid-js';
example[0];"
`;
